cpp/src/table/row_operators.cu

```diff
@@ -402,126 +483,119 @@ namespace {
  * If the input column is not lists-of-structs, or does not contain lists-of-structs at any nested
  * level, the input will be passed through without any changes.
  *
- * @param lhs The input lhs column to transform
- * @param rhs The input rhs column to transform (if available)
+ * @param input The input column to transform
  * @param column_null_order The flag indicating how nulls compare to non-null values
  * @param stream CUDA stream used for device memory operations and kernel launches
- * @return A tuple consisting of new column_view representing the transformed input, along with
- *         their ranks column(s) (of `size_type` type) and possibly new list offsets generated
+ * @param mr Device memory resource used to allocate the returned column(s)
+ * @return A pair consisting of new column_view representing the transformed input, along with
+ *         an array containing its rank column(s) (of `size_type` type) and possibly new list
+ *         offsets generated during the transformation process
+ */
+std::pair<column_view, std::vector<std::unique_ptr<column>>> transform_lists_of_structs(
+  column_view const& input,
+  null_order column_null_order,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  std::vector<std::unique_ptr<column>> out_cols;
+
+  if (input.type().id() == type_id::LIST) {
+    auto const child = cudf::lists_column_view{input}.get_sliced_child(stream);
+
+    // Found a lists-of-structs column.
+    if (child.type().id() == type_id::STRUCT) {
+      out_cols.emplace_back(compute_ranks(child, column_null_order, stream, mr));
+      auto transformed = replace_child(input, out_cols.back()->view(), out_cols, stream, mr);
+      return {std::move(transformed), std::move(out_cols)};
```

Comment:
If you directly put the `replace_child` call into the pair, you can remove the `std::move` and rely on RVO right?

Reply:
@ttnghia I see that this conversation was resolved but no action was taken. Was that intentional because it doesn't work, or do you have unpushed local changes?

Reply:
Sorry I made changes but forgot to push :smile:

Reply:
I figured :slightly_smiling_face: no worries